pennylane/qnode.py

```diff
@@ -692,6 +740,12 @@ def _pd_analytic(self, params, idx, force_order2=False, **kwargs):
         Returns:
             float: partial derivative of the node.
         """
+        # remove jacobian specific keyword arguments
```

Comment:
seems like we might want to put in a small helper function for this, since it is duplicated elsewhere

Reply:
This is just a short term solution though, might be better to just bite the bullet and implement a `grad_options` dictionary. A small change, but it will mean updating a lot of the tests in `test_quantum_gradients` and `test_qnode`.

Reply:
yep, we might do that. I was thinking of removing duplicated code even now, even if it will end up getting trashed in a future PR anyway

Reply:
done